clients/hadoopfs/src/main/java/io/lakefs/FileSystemTracer.java

```diff
@@ -0,0 +1,353 @@
+        s3AFileSystem = s3aPath.getFileSystem(conf);
+
+        useLakeFSFileSystemRes = conf.getBoolean(USE_LAKEFS_RESPONSE, false);
+        LOG.trace("Initialized FileSystemTracer, fs.lakefs.tracer.use.lakefs: {}", useLakeFSFileSystemRes);
+    }
+
+    @Override
+    public URI getUri() {
+        LOG.trace("getUri");
+
+        URI lakefsRes = lfsFileSystem.getUri();
+        URI s3aRes = s3AFileSystem.getUri();
+        LOG.trace("{}[getUri] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public Path makeQualified(Path path) {
+        LOG.trace("makeQualified");
+
+        Path lakefsRes = lfsFileSystem.makeQualified(path);
+        Path s3aRes = s3AFileSystem.makeQualified(translateLakeFSPathToS3APath(path));
+        LOG.trace("{}[makeQualified] lakefs: {}, s3a: {}",RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return translateS3APathToLakeFSPath(s3aRes);
+    }
+
+    @Override
+    public FSDataInputStream open(Path f, int bufferSize) throws IOException {
+        LOG.trace("open(Path {}, bufferSize {})", f, bufferSize);
+
+        FSDataInputStream lakefsRes = lfsFileSystem.open(f, bufferSize);
+        FSDataInputStream s3aRes = s3AFileSystem.open(translateLakeFSPathToS3APath(f), bufferSize);
+        LOG.trace("{}[open]: lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public FSDataOutputStream create(Path f, FsPermission permission, boolean overwrite, int bufferSize,
+                                     short replication, long blockSize, Progressable progress) throws IOException {
+        LOG.trace("create(Path {}, permission {}, overwrite {}, bufferSize {}, replication {}, blockSize {}, progress {})",
+                f, permission, overwrite, bufferSize, replication, blockSize, progress);
+
+        TracerOutputTStream tOutputStream = new TracerOutputTStream(f, permission, overwrite, bufferSize, replication, blockSize, progress);
+        return new FSDataOutputStream(tOutputStream, null);
+    }
+
+    @Override
+    public FSDataOutputStream append(Path f, int bufferSize, Progressable progress) throws IOException {
+        LOG.trace("append(f {}, bufferSize {}, progress {})", f, bufferSize, progress);
+
+        FSDataOutputStream lakefsRes = lfsFileSystem.append(f, bufferSize, progress);
+        FSDataOutputStream s3aRes = s3AFileSystem.append(translateLakeFSPathToS3APath(f), bufferSize, progress);
+        LOG.trace("{}[append] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;//unused
```

Comment:
depends on the configuration - or old comment

Reply:
old comment, removed it